### ISSUE TITLE
Error Messages and Red Lights Displayed in IntelliJ IDEA Editor although test was fixed

### ISSUE THREAD

**Issue by pahund** [wallabyjs/public#570]
### Issue description or question

When I'm working on a test and doing some refactoring, the refactoring in progress results in a compile error. Wallaby displays this error next to the code that I'm working on and shows a red light at the beginning of the line. So far so good, that is expected.

However, when the refactoring is finished and the code compiles again, the error message in the editor stays and the line continues to be marked red, while Wallaby tool window correctly reports “no failing tests”.

I have to restart my Wallaby.js run configuration to get a green light on the refactored code line.

I made a video to demonstrate the bug: https://youtu.be/0woztRC_Kl0

### Wallaby.js configuration file

```javascript
module.exports = wallaby => ({
    files: [
        "frontend/app/**/*.js",
        "frontend/app/**/*.scss",
        "frontend/test/testdata.json",
        "frontend/test/testUtils.js"
    ],

    tests: [
        "frontend/test/**/*Test.js"
    ],

    env: {
        type: "node",
        runner: "/usr/local/bin/node"
    },

    testFramework: "mocha",

    compilers: {
        "**/*.js": wallaby.compilers.babel()
    },

    bootstrap: () => {
        const path = require("path");
        const setupPath = path.join(wallaby.localProjectDir, "frontend/test/setup");
        require(setupPath);
    }
});
```

### Code editor or IDE name and version

IntelliJ IDEA Ultimate 2016.1.1

### OS name and version

OS X El Capitan 10.11.4

**Comment by ArtemGovorov** [comment]
Thanks for the detailed description and the video! 
We have had a similar issue reported internally some time ago and IIRC it was related to an issue in Babel with generating correct source maps. What Babel version are you using?

**Comment by ArtemGovorov** [comment]
Quick note, as a faster workaround for glitches like this one, you may hit `Alt + Enter` and choose `Run file tests` instead of restarting wallaby.js.

**Comment by ArtemGovorov** [comment]
Could you please also share your `.babelrc` file?

**Comment by ArtemGovorov** [comment]
Another workaround for the issue is to use block statement:
```javascript
it('has a child <h3> element', () => {
  childOf...
});
```
instead of the arrow function expression:
```javascript
it('has a child <h3> element', () => 
  childOf...
);
```

**Comment by pahund** [comment]
Hi @ArtemGovorov, thanks for the quick reply!
I'm using Babel 6.7.5, my .babelrc looks like this:
````json
{
  "presets": ["es2015", "react"]
}
````

**Comment by pahund** [comment]
In my webpack config, I use devtool "#source-map", but webpack is not involved in my test setup. I'm assuming the source maps come from some config inside Wallaby?

**Comment by ArtemGovorov** [comment]
Babel generates source maps when compiling you files, wallaby just uses those maps to map error stack from the actually executed compiled code back to your original code. There was an issue with babel generated source maps that was causing a similar issue.

**Comment by pahund** [comment]
So could I work around by configuring wallaby to use a different type of source map, e.g. inline? How would I do that?

**Comment by ArtemGovorov** [comment]
No, the source map type doesn't matter. There's no workaround that can be done with some configuration. You may use [this workaround for now](https://github.com/wallabyjs/public/issues/570#issuecomment-215394337).

**Comment by pahund** [comment]
OK, thanks

**Comment by ArtemGovorov** [comment]
After a careful investigation, it turned out to be an issue in wallaby.js, not in babel generated source maps (even though it's is related to the way babel generates source maps for arrow functions with no body).

The issue is fixed in wallaby core v1.0.223 (you may restart your IDE and start wallaby to force the update). Thanks again for the detailed description and the issue video, it was very helpful in finding out the reason and fixing it.

**Comment by pahund** [comment]
Thanks for acting so quickly! Cheers!
